pkg/controller/certificates/revisionmanager/revisionmanager_controller.go

```diff
@@ -117,9 +117,15 @@ func (c *controller) ProcessItem(ctx context.Context, key types.NamespacedName)
 
 	log = logf.WithResource(log, crt)
 
-	// If RevisionHistoryLimit is nil, don't attempt to garbage collect old
-	// CertificateRequests
+	// If RevisionHistoryLimit is nil, then default to 1
 	if crt.Spec.RevisionHistoryLimit == nil {
+		defaultRevisionHistoryLimit := int32(1)
+		crt.Spec.RevisionHistoryLimit = &defaultRevisionHistoryLimit
```

Comment:
We probably don't want to modify the resource (`crt.Spec.RevisionHistoryLimit`), this might cause concurrency issues in our informer cache. Instead, we should modify the limit variable below.